vnext/Shared/Hasher.cpp

```diff
@@ -0,0 +1,78 @@
+#include "pch.h"
+#include "Hasher.h"
+
+#ifndef NT_SUCCESS
+#define NT_SUCCESS(Status) ((NTSTATUS)(Status) >= 0)
+#endif // NT_SUCCESS
+
+namespace Microsoft::ReactNative {
+
+  SHA256Hasher::SHA256Hasher()
+  {
+    NTSTATUS status;
+    if (!NT_SUCCESS(status = BCryptOpenAlgorithmProvider(&m_hAlg, BCRYPT_SHA256_ALGORITHM, NULL, 0)))
+    {
+      // Failed to load and initialize a CNG provider
+      throw status;
+    }
+
+    DWORD cbObject;
+    DWORD cbData;
+    if (!NT_SUCCESS(status = BCryptGetProperty(m_hAlg, BCRYPT_OBJECT_LENGTH, (PBYTE)&cbObject, sizeof(DWORD), &cbData, 0)))
+    {
+      // Failed to retrieve the value for BCRYPT_OBJECT_LENGTH
+      throw status;
+    }
+
+    m_hashObject.resize(cbObject);
+    if (!NT_SUCCESS(status = BCryptCreateHash(m_hAlg, &m_hHash, m_hashObject.data(), static_cast<ULONG>(m_hashObject.size()), NULL, 0, 0)))
+    {
+      // Failed to create hash
+      throw status;
+    }
+  }
+
+  SHA256Hasher::~SHA256Hasher()
+  {
+    BCryptCloseAlgorithmProvider(m_hAlg, 0);
+    BCryptDestroyHash(m_hHash);
```

Comment:
>BCryptDestroyHash [](http://example.com/codeflow?start=4&length=17)

I wonder about the destruction order: should we destroy the hash before closing algorithm? I understand that the Hash points to the algorithm.